lib/control.ml

```diff
@@ -37,7 +37,6 @@ let unix_timeout n f x =
   (* Here we assume that the existing timer will also interrupt us. *)
   if old_timer.it_value > 0. && old_timer.it_value <= n then Some (f x) else
     let psh = Sys.signal Sys.sigalrm (Sys.Signal_handle timeout_handler) in
-    let old_timer = setitimer ITIMER_REAL {it_interval = 0.; it_value = n} in
     let restore_timeout () =
       let timer_status = getitimer ITIMER_REAL in
       let old_timer_value = old_timer.it_value -. n +. timer_status.it_value in
```

Comment:
This changes what old_timer is bound to here, is it still correct?

Reply:
Indeed I think it is not, `old_timer` should be now a parameter of restore_timeout.

Reply:
My analysis is as follows:
- Both `getitimer` on line 36 and `setitimer` return the existing setting of the timer. In the old code they both bound this to `old_timer`, I will refer to them as `oldtimer1` and `oldtimer2`.
- It holds that `oldtimer1.it_value = oldtimer2.it_value - epsilon` where `epsilon` is the time it took to execute the lines 38-39.
- `epsilon` is negligible (smaller than timeout precision anyway).
- Hence, technically speaking, the semantics are changed, practically speaking not.

That being said if you guys like it more, I'll be happy to move `restore_timeout` into the `try` so it an reference the second binding of `old_timer`.

Reply:
(Oh, I cannot move `restore_timeout` into the `try`, because it is used in the `with`. So like @ejgallego says, I'd have to parameterize it. But I don't think that is better because then we will sometimes restore the timeout from `oldtimer1` and sometimes from `oldtimer2`, which is a bit strange I guess.)